homeassistant/components/mpd/media_player.py

```diff
@@ -222,12 +223,20 @@ def source(self):
 
     @property
     def source_list(self):
-        """Return the list of available input sources."""
+        """Return the list of songs or list of playlists."""
+        if self._currentplaylistsongs:
+            return self._currentplaylistsongs
```

Comment:
Only input sources should be returned by the `source_list` property, as already pointed out.

See this comment for our plan on how to extend the media_player:
https://github.com/home-assistant/architecture/issues/62#issuecomment-418289560

That plan is just a plan so far.